src/components/TodoList.tsx

```diff
@@ -0,0 +1,41 @@
+import { memo } from 'react';
+import { Todo } from '../types/Todo';
+import { TodoItem } from './TodoItem';
+
+interface Props {
+  todos: Todo[],
+  tempTodo: Todo | null,
+  deleteTodo: (id: number) => void,
+  loadingTodoIds: number[],
+  updateTodo: (id: number, data: Partial<Todo>) => void,
+}
+
+export const TodoList: React.FC<Props> = memo(({
+  todos, tempTodo, deleteTodo, loadingTodoIds, updateTodo,
+}) => {
+  return (
+    <section className="todoapp__main" data-cy="TodoList">
```

Comment:
Use semantic tags for list `ul` `li`. Don't forget to remove default `li` dots and `ul` paddings